src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -260,6 +260,7 @@ def __init__(
             referenced_ctes.copy() if referenced_ctes else set()
         )
         self._cumulative_node_complexity: Optional[Dict[PlanNodeCategory, int]] = None
+        self._uuid = str(uuid.uuid4())
```

Comment:
so there is very embracing state here that the snowflake_plan its-self already have and self._id fields, and used as part of the __eq__ comparison, the id is generated based on the query and params, which may not be the exact indicator of whether two plan are the same plan. 
I think we can keep both, but let's make sure we we doc it clearly here about the difference. @sfc-gh-jdu i also start question is changing the __eq__ and __hash__ correct in this case? I got the point that we want to simplify the comparison, but that also kind of changes the meaning of the equivalence of two snowflake plan nodes here.